agent/src/softwarecontaineragent.cpp

```diff
@@ -91,6 +94,47 @@ SoftwareContainerAgent::~SoftwareContainerAgent()
 {
 }
 
+/* we are fetching lxcpath from config now. But in near future we should use lxcConfigPath instead
+ * */
+void SoftwareContainerAgent::removeOldContainers()
+{
+    char **containerNames = nullptr;
+    struct lxc_container **containerList = nullptr;
+
+    const char *basePath = lxc_get_global_config_item("lxc.lxcpath");
+    auto num = list_all_containers(basePath, &containerNames, &containerList);
+    if (num > 0 && num < std::numeric_limits<int>::max()) {
```

Comment:
if we are trusting the library that we are sure that `list_all_containers` only return -1  on error and count of containers on success, then you are right. This check is questioning borders instead of error check, which is safer. I didn't look in `list_all_containers` code so i don't know whether it has checks or not.